h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -264,14 +268,21 @@ public AutoBuffer( InputStream is ) {
     _bb.flip();
     _is = is;
     int b = get1U();
-    if( b==0 ) return;          // No persistence info
-    int magic = get1U();
-    if( b!=0x1C || magic != 0xED ) throw new IllegalArgumentException("Missing magic number 0x1CED at stream start");
-    checkVersion(getStr());
-    String[] typeMap = getAStr();
+
+    if (typeMap == null) {
+      if (b == 0)
+        return;          // No persistence info
+      int magic = get1U();
+      if (b != 0x1C || magic != 0xED) throw new IllegalArgumentException("Missing magic number 0x1CED at stream start");
+      checkVersion(getStr());
+      typeMap = getAStr();
+      assert typeMap != null;
+    } else {
+      assert b == 0;
```

Comment:
Isn't it better to throw a proper exception identifying that the content is malformed for the current use case?

Reply:
you are right, this should be improved - thanks for the suggestion